tests/test_subprocess.py

```diff
@@ -0,0 +1,91 @@
+"""`pytest` tests for utils/subprocess.py"""
+
+import subprocess
+import pytest
+
+from benchcab.utils.subprocess import SubprocessWrapper
+from .common import TMP_DIR
+
+
+def test_run_cmd(capfd):
```

Comment:
I don't think it is useless: we were printing the command any time we call `subprocess.run()` in verbose mode, so I think that was why we ended up putting the print statement inside the `run_cmd()` function. Can you elaborate on what is weird about the tests?

Reply:
Why I think it is useless is because in the new implementation currently, the 'cmd' is an input argument to run_cmd and goes straight into the subprocess call. The 'cmd' is actually set by the calling method. And we don't use that in any way to ensure the 'cmd' variable is still set correctly. To test that we mock the subprocess call and check the commands that is passed as argument directly and those tests are done for each of the calling methods.

Why the tests are weird: when you have a test that says you are going to pass the command "echo foo" to subprocess and check the std output is printed, you would expect to get "foo\n" not "echo foo\nfoo\n". The simple solution is a better comment that reminds the reader the function outputs the cmd and the output of the function in subprocess when it is verbose.

But as I said, that's rather a detail and probably best to leave it alone for now.

Reply:
> Why I think it is useless is because in the new implementation currently, the 'cmd' is an input argument to run_cmd and goes straight into the subprocess call. The 'cmd' is actually set by the calling method. And we don't use that in any way to ensure the 'cmd' variable is still set correctly. To test that we mock the subprocess call and check the commands that is passed as argument directly and those tests are done for each of the calling methods.

I see... The reason why we print the command here is so that the user knows what is being run (this was added in PR https://github.com/CABLE-LSM/benchcab/pull/69). So in our unit tests, we are not necessarily testing the command being run is correct but that we are testing the user can see the command being run in the standard output.

> Why the tests are weird: when you have a test that says you are going to pass the command "echo foo" to subprocess and check the std output is printed, you would expect to get "foo\n" not "echo foo\nfoo\n". The simple solution is a better comment that reminds the reader the function outputs the cmd and the output of the function in subprocess when it is verbose.

I have added better comments that describe the test here: [95f1871](https://github.com/CABLE-LSM/benchcab/pull/99/commits/95f187105c25613915dc49b0185b4aebefff0f42)